sunpy/map/sources/suit.py

```diff
@@ -0,0 +1,121 @@
+"""
+SUIT Map sub class definitions
+
+"""
+from sunpy.map.sources.source_type import source_stretch
+import numpy as np
+import astropy.units as u
+from astropy.coordinates import CartesianRepresentation, HeliocentricMeanEcliptic
+from sunpy.map.mapbase import GenericMap
+from matplotlib import colors
+from astropy.visualization import ImageNormalize
+from astropy.visualization import AsinhStretch
+
+__all__ = ['SUITMap']
+__author__ = ['Rahul Gopalakrishnan']
+__email__ = ['rahulg.astro@gmail.com']
+
+class SUITMap(GenericMap):
+    """
+    The Solar Ultraviolet Imaging Telescope (SUIT) is one of the remote sensing payloads on board the Aditya-L1 mission of the Indian Space Research Organization (ISRO)
+    that was launched on September 02, 2023. SUIT is designed to provide near-simultaneous full-disk and region-of-interest images of the Sun at various heights,
+    slicing through the photosphere and chromosphere, employing an array of 11 scientifically calibrated filters (3 broad-band & 8 narrow-band) strategically
+    positioned within the wavelength range of 200 to 400 nanometers. Located at the first Lagrange point, SUIT observes the Sun 24x7, without any interruption.
+
+    References:
+    -----------
+    * `Home Page: <https://suit.iucaa.in/>`_
+    * `Mission Page: <https://suit.iucaa.in/about_SUIT>`_
+    * `Publications Page: <https://suit.iucaa.in/node/5>`_
+    """
+
+    def __init__(self, data, header, **kwargs):
+        super().__init__(data, header, **kwargs)
+        self._nickname = self.detector
+        self.filter = header.get('FTR_NAME').strip()
+        self.wavelnth = str(header.get('WAVELNTH','NA')).strip()
+        self.plot_settings['cmap'] = f"suit_{self.filter.lower()}"
+        self.plot_settings['title'] = f"SUIT {self.filter}:{self.wavelnth} A - {self.reference_date}"
+        norm = self.suit_norm()
+
+        if self.filter in norm:
+            self.plot_settings["norm"] = norm[self.filter]
+        else:
+            self.plot_settings["norm"] = ImageNormalize(
+                stretch=source_stretch(self.meta, AsinhStretch(0.01)), clip=False
+            )
+
+    def suit_norm(self):
+        '''
+        Function to fix a norm for the image from its median value
+        '''
+        a = np.array(self.data)
+        sliced_array = a[(a > 2000) & (a < 50000)]
+        baseline = np.median(sliced_array) if len(sliced_array) > 0 else 1000
+        norm_params = {
```

Comment:
Our rule for normalising Maps is that the default shouldn't clip data, we use stretches instead.

For more details you can see the discussion in this PR where it came up last time: https://github.com/sunpy/sunpy/pull/7180#pullrequestreview-1677077264